cmd/internal/apidump/apidump.go

```diff
@@ -44,15 +47,56 @@ var (
 	maxWitnessSize_bytes    int
 	dockerExtensionMode     bool
 	healthCheckPort         int
+	randomizedStart         int
 )
 
+// This function will either startup apidump normally, or never return, with probability
+// determined by randomizedStart/100.  The value of the command-line flag may be
+// overridden by an environment variable to make it easier to apply.
+// This function should be called as early as possible in case termination of
+// the agent causes deployment problems.
+//
+// Negative values are effectively treated as 0 probability, instead of being validated.
+func applyRandomizedStart() {
+	prob := randomizedStart
+
+	if env := os.Getenv("POSTMAN_AGENT_RANDOM_START"); env != "" {
+		override, err := strconv.Atoi(env)
+		if err == nil {
+			prob = override
+		}
+	}
+
+	if prob < 100 {
+		printer.Stdout.Infof("Starting trace with probability %d%%.\n", prob)
+
+		// Pre-1.20, Go does not seed the default Random object :(
+		rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+		r := rng.Intn(100) // in range [0,100),
+		// so 1% probability means < 1, not <= 1:
+		if r < prob {
+			return
+		}
+
+		printer.Stdout.Infof("This agent instance will not begin capturing data.\n")
+
+		// Wait forever
+		select {}
+
+		// Unreachable
+		os.Exit(0)
```

Comment:
When will this be called?

Reply:
Never, that's why it's marked unreachable (and the linter will tell you so too.)

I just felt it was more robust to have it here in case the select was later modified.